earth_enterprise/src/fusion/autoingest/sysman/AssetVersionD.cpp

```diff
@@ -91,11 +92,12 @@ AssetVersionImplD::StateChangeNotifier::NotifyParents(
     std::shared_ptr<StateChangeNotifier> notifier) {
   notify(NFY_VERBOSE, "Iterate through parents");
   int i = 1;
-  for (const std::string & ref : parentsToNotify) {
+  std::vector<std::string> theRefs = parentsToNotify.keys();
```

Comment:
There's a potential issue here where we call `parentsToNotify.clear()` at the end of this function - that could clear out things that were added after you called `keys()`. On the other hand, this function is only called from the destructor, so it's guaranteed to run in a single thread anyway. (On this branch it may be called in 2 places, but the one that's not in the destructor has been removed on @tst-sbayard's version of this branch).